### PR TITLE
Swimm UI Next

### PR DESCRIPTION
**Pull request by talkor**
- Upgrade to Vue 3
- Use Vite 
- Bundle Component Library with Rollup
- Add Prettier
- Upgrade to Vuepress 2
- Create a component demo plugin for Vuepress 2 and use it
- Publish Vuepress to GitHub pages
- Rearrange imports in docs
- Migrate from Jest to Vitest
- Minor bug fixes
- Remove old app and unused files
- Change CSS Modules to Scoped CSS
- Prepare library for automated publish using semantic-release (different PR to come)

### CODE REVIEW

File: .github/workflows/publish_docs.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            ${{ runner.os }}-yarn-
+
+      - name: Install dependencies
+        if: steps.yarn-cache.outputs.cache-hit != 'true'
+        run: yarn --frozen-lockfile
+
+      - name: Build VuePress site
+        run: yarn docs:build
+
+      - name: Deploy to GitHub Pages
```
Comment (talkor) [inline .github/workflows/publish_docs.yml:39]: Check action from github

File: package.json [hunk 2]
```diff
@@ -1,50 +1,42 @@
...
+    "vite": "^2.9.5",
+    "vitest": "^0.9.3",
+    "vuepress": "^2.0.0-beta.43"
   },
   "peerDependencies": {
-    "vue": "^2.6.12"
-  },
-  "engines": {
-    "node": ">=12"
+    "vue": "^3.2.33"
```
Comment (talkor) [inline package.json:40]: > 3

File: .github/workflows/publish_docs.yml (hunk 1 above)
Comment (talkor) [reply .github/workflows/publish_docs.yml:39]: Recommanded by Vuepress

**Review by YossiSaadi — APPROVED** [review]
Amazing job, shapo 🎉 👏🏼

**Pull request merged**